wagtail/fields.py

```diff
@@ -111,6 +111,10 @@ def __init__(self, block_types, use_json_field=True, **kwargs):
 
         self.stream_block.set_meta_options(block_opts)
 
+    @property
+    def value_class(self):
+        return self.stream_block.value_class
```

Comment:
in `__init__` we have:

```python
if isinstance(block_types, Block):
    # use the passed block as the top-level block
    self.stream_block = block_types
```

but StreamField really expectes `BaseStreamBlock` (which inherits from `Block`). I wonder if the check should change to `BaseStreamBlock`, or perhaps this to use `getattr`.

I couldn't find any tests of what happens if you try to initialize StreamField with a non-StreamBlock block. Perhaps I am overthinking this

Reply:
> I couldn't find any tests of what happens if you try to initialize StreamField with a non-StreamBlock block. Perhaps I am overthinking this

This isn't currently supported (but might be in future, once #9746 lands).